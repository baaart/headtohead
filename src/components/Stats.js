import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Stat from './Stat';

const StatWrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0 2em;
  width: 100%;
`

function Stats({...props}) {
  return (
    <StatWrapper>
      <h3>Stats</h3>
      <Stat label="Rank" value={props.data.rank} />
      <Stat label="Points" value={props.data.points} />
      <Stat label="Wins" value={props.data.last} formatter={v => v.filter(x => x===1).length} />
      <Stat label="Loses" value={props.data.last} formatter={v => v.filter(x => x===0).length} />
      <Stat label="Height" value={props.data.height} formatter={v => `${v}cm`} />
      <Stat label="Weight" value={props.data.weight} formatter={v => `${v/1000}kg`}/>
    </StatWrapper>
  );
}

Stats.propTypes = {
  data: PropTypes.object.isRequired
};

export default Stats;
