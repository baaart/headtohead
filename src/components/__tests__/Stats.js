import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Stats from '../Stats';
import { fakeData } from '../__mocks__/fakeData'

describe('Stats', () => {
  it('Matches snapshot', () => {
    const wrapped = shallow(<Stats data={fakeData.players[0].data} />);
    console.log(wrapped.debug());
    expect(wrapped).toMatchSnapshot();
  })
});
