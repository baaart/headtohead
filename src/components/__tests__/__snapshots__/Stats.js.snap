// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Stats
    data={
      Object {
        "age": 33,
        "height": 183,
        "last": Array [
          1,
          1,
          1,
          0,
          1,
        ],
        "points": 1784,
        "rank": 21,
        "weight": 81000,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3>
          Stats
        </h3>,
        <Stat
          formatter={[Function]}
          label="Rank"
          value={21}
        />,
        <Stat
          formatter={[Function]}
          label="Points"
          value={1784}
        />,
        <Stat
          formatter={[Function]}
          label="Wins"
          value={
            Array [
              1,
              1,
              1,
              0,
              1,
            ]
          }
        />,
        <Stat
          formatter={[Function]}
          label="Loses"
          value={
            Array [
              1,
              1,
              1,
              0,
              1,
            ]
          }
        />,
        <Stat
          formatter={[Function]}
          label="Height"
          value={183}
        />,
        <Stat
          formatter={[Function]}
          label="Weight"
          value={81000}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Stats",
        },
        "ref": null,
        "rendered": "Stats",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "formatter": [Function],
          "label": "Rank",
          "value": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "formatter": [Function],
          "label": "Points",
          "value": 1784,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "formatter": [Function],
          "label": "Wins",
          "value": Array [
            1,
            1,
            1,
            0,
            1,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "formatter": [Function],
          "label": "Loses",
          "value": Array [
            1,
            1,
            1,
            0,
            1,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "formatter": [Function],
          "label": "Height",
          "value": 183,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "formatter": [Function],
          "label": "Weight",
          "value": 81000,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-direction: column;
  margin: 0 2em;
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3>
            Stats
          </h3>,
          <Stat
            formatter={[Function]}
            label="Rank"
            value={21}
          />,
          <Stat
            formatter={[Function]}
            label="Points"
            value={1784}
          />,
          <Stat
            formatter={[Function]}
            label="Wins"
            value={
              Array [
                1,
                1,
                1,
                0,
                1,
              ]
            }
          />,
          <Stat
            formatter={[Function]}
            label="Loses"
            value={
              Array [
                1,
                1,
                1,
                0,
                1,
              ]
            }
          />,
          <Stat
            formatter={[Function]}
            label="Height"
            value={183}
          />,
          <Stat
            formatter={[Function]}
            label="Weight"
            value={81000}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Stats",
          },
          "ref": null,
          "rendered": "Stats",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "formatter": [Function],
            "label": "Rank",
            "value": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "formatter": [Function],
            "label": "Points",
            "value": 1784,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "formatter": [Function],
            "label": "Wins",
            "value": Array [
              1,
              1,
              1,
              0,
              1,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "formatter": [Function],
            "label": "Loses",
            "value": Array [
              1,
              1,
              1,
              0,
              1,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "formatter": [Function],
            "label": "Height",
            "value": 183,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "formatter": [Function],
            "label": "Weight",
            "value": 81000,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: column;
  margin: 0 2em;
  width: 100%;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
